### PR TITLE
OCM-9892 | feat: add root disk size to TF provider

### PR DESCRIPTION
**Pull request by willkutler**
**What this PR does / why we need it**:
- Adds the custom disk size field for creating node pools to the terraform provider, so that node pools with a custom disk size can be created through terraform
- Adds a disk size field to HCP cluster creation so that day 1 node pools w/ custom disk size can be created through terraform

**Which issue(s) this PR fixes** *(optional, use `fixes #<issue_number>(, fixes #<issue_number>, ...)` format, where issue_number might be a GitHub issue, or a Jira story (OCM-xxxx)*:
https://issues.redhat.com/browse/OCM-9892

**Change type**
- [ x] New feature
- [ ] Bug fix
- [ ] Build
- [ ] CI
- [ ] Documentation
- [ ] Performance
- [ ] Refactor
- [ ] Style
- [ ] Unit tests
- [ x] Subsystem tests

**Checklist**
- [ ] Subject and description added to both, commit and PR.
- [ ] Relevant issues have been referenced.

### CODE REVIEW

File: provider/machinepool/hcp/machine_pool_resource.go [hunk 1]
```diff
@@ -310,6 +310,14 @@ func (r *HcpMachinePoolResource) Create(ctx context.Context, req resource.Create
...
 		}
 		awsNodePoolBuilder.Ec2MetadataHttpTokens(cmv1.Ec2MetadataHttpTokens(plan.AWSNodePool.Ec2MetadataHttpTokens.ValueString()))
+
+		if common.HasValue(plan.AWSNodePool.DiskSize) {
+			awsNodePoolBuilder.RootVolume(cmv1.NewAWSVolume().Size(int(plan.AWSNodePool.DiskSize.ValueInt64())))
+		}
+		if workerDiskSize := common.OptionalInt64(plan.AWSNodePool.DiskSize); workerDiskSize != nil {
+			awsNodePoolBuilder.RootVolume(cmv1.NewAWSVolume().Size(int(*workerDiskSize)))
+		}
+
```
Comment (marcolan018) [inline provider/machinepool/hcp/machine_pool_resource.go:320]: why do we check plan.AWSNodePool.DiskSize twice here?

File: provider/machinepool/hcp/machine_pool_resource.go [hunk 2]
```diff
@@ -595,6 +603,7 @@ func validateNoImmutableAttChange(state, plan *HcpMachinePoolState) diag.Diagnos
 			"aws_node_pool.additional_security_group_ids", &diags)
 		validateStateAndPlanEquals(state.AWSNodePool.Ec2MetadataHttpTokens, plan.AWSNodePool.Ec2MetadataHttpTokens,
 			"aws_node_pool.ec2_metadata_http_tokens", &diags)
+		validateStateAndPlanEquals(state.AWSNodePool.DiskSize, plan.AWSNodePool.DiskSize, "disk_size", &diags)
```
Comment (marcolan018) [inline provider/machinepool/hcp/machine_pool_resource.go:606]: "disk_size" should be "aws_node_pool.disk_size"

File: provider/clusterrosa/hcp/resource.go [hunk 3]
```diff
@@ -364,6 +364,10 @@ func (r *ClusterRosaHcpResource) Schema(ctx context.Context, req resource.Schema
 					stringplanmodifier.UseStateForUnknown(),
 				},
 			},
+			"worker_disk_size": schema.Int64Attribute{
```
Comment (marcolan018) [inline provider/clusterrosa/hcp/resource.go:367]: if this one cannot be updated after cluster creation, should check it in validateNoImmutableAttChange() func

**Comment by 86254860** [comment]
@willkutler Do you know where do we usually update the examples for this new update?

**Comment by 86254860** [comment]
@willkutler Could you also take a look about the failed test job, it raise the  below error, which seems that it doesn't support the scenario that user don't configure `disk_size` 
```
Mismatch between struct and object type: Struct defines fields not found in
      object: disk_size.
      Struct: hcp.AWSNodePool
```
https://storage.googleapis.com/test-platform-results/pr-logs/pull/terraform-redhat_terraform-provider-rhcs/793/pull-ci-terraform-redhat-terraform-provider-rhcs-main-rosa-hcp-advanced-critical-high-presubmit/1836496582241947648/build-log.txt

**Comment by willkutler** [comment]
@marcolan018 do you have any idea what would be causing this test failure? The error message seems like the problem is the disk size not being defined in the AWSNodePool struct but I have added the field to that struct in `provider/machinepool/hcp/aws_node_pool.go`

File: provider/clusterrosa/hcp/resource.go [hunk 4]
```diff
@@ -365,6 +365,10 @@ func (r *ClusterRosaHcpResource) Schema(ctx context.Context, req resource.Schema
...
+				Description: "Compute node root disk size, in GiB. " + rosaTypes.PoolMessage,
+        Optional:    true,
+      },
```
Comment (gdbranco) [inline provider/clusterrosa/hcp/resource.go:371]:
```suggestion
				Description: "Compute node root disk size, in GiB. " + rosaTypes.PoolMessage,
                                Optional:    true,
      },
```

**Pull request merged**